hail/python/hailtop/aiocloud/aiogoogle/client/storage_client.py

```diff
@@ -630,22 +635,28 @@ def schemes() -> Set[str]:
     def storage_location(self, uri: str) -> str:
         return self.get_bucket_and_name(uri)[0]
 
-    async def is_hot_storage(self, location: str) -> bool:
+    async def is_hot_storage(self, location: str, uri: str) -> bool:
```

Comment:
What should happen if `uri` is a directory?

Reply:
Ya I mean "directory" here loosely in the sense that object storage try to look like actual filesystems. Indeed there does not appear to be an obviously correct solution in this case, but unfortunately I think that directories are going to be a common use case. Since hail datasets are partitioned across multiple files, any `.ht` or `.mt` is actually a directory. You can see for example the gnomad public data: [gs://gcp-public-data--gnomad/release/4.0/ht/genomes/gnomad.genomes.v4.0.sites.ht](https://console.cloud.google.com/storage/browser/gcp-public-data--gnomad/release/4.0/ht/genomes/gnomad.genomes.v4.0.sites.ht?pageState=(%22StorageObjectListTable%22:(%22f%22:%22%255B%255D%22))&prefix=&forceOnObjectsSortingFiltering=false).

> if there's a way for us to list objects in it i guess we could just check the first one we see?

Regardless though, I think this is a very reasonable solution. If someone gives us a directory it is most often a directory of data that belong together, like a Hail dataset. I am fine expecting those all to be the same storage type.

Reply:
updated!